### PR TITLE
Feature/di for tg webapi added

### PR DESCRIPTION
**Pull request by gitrepository100**
State machine interfaces/base classes added

### CODE REVIEW

File: OneSchedule.Domain.Abstractions/IStateMachineState.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace OneSchedule.Domain.Abstractions
+{
+    public interface IStateMachineState
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/IStateMachineState.cs]: Put classes related to State machine implementation to a separate folder. After that rename IStateMachineState interface to IState and update inheritors' names accordingly

File: OneSchedule.Domain.Abstractions/IStateMachineState.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        public void Handle();
+        public void SetContext(IStateMachineContext context);
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/IStateMachineState.cs]: Add empty line between methods

File: OneSchedule.Domain.Abstractions/OneSchedule.Domain.Abstractions.csproj [hunk 3]
```diff
@@ -1,11 +1,12 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="MongoDB.Driver" Version="2.13.1" />
+    <PackageReference Include="Telegram.Bot" Version="16.0.2" />
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/OneSchedule.Domain.Abstractions.csproj]: Why was Telegram bot library added to Abstractions project?

File: OneSchedule.Domain/StateMachine/StateMachineContext.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+using System.Threading.Tasks;
+using Telegram.Bot;
+using Telegram.Bot.Types;
+
+namespace OneSchedule.Domain
+{
+    public class StateMachineContext : IStateMachineContext
+    {
+        private IStateMachineState _state;
+        public ITelegramBotClient Bot;
```
Comment (elkavitaly) [inline OneSchedule.Domain/StateMachine/StateMachineContext.cs]: Why are these fields public?

File: OneSchedule.Domain/StateMachine/StateMachineState.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+﻿
```
Comment (elkavitaly) [inline OneSchedule.Domain/StateMachine/StateMachineState.cs]: Remove empty line

File: OneSchedule.Domain/StateMachine/StateMachineState.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot;
+using OneSchedule;
+
+namespace OneSchedule.Domain
+{
+    public class StateMachineState : IStateMachineState
```
Comment (elkavitaly) [inline OneSchedule.Domain/StateMachine/StateMachineState.cs]: What is the purpose of this class? Does it represent some specific state? If yes, please, specify which state exactly it represents.

File: OneSchedule.Domain/StateMachine/StateMachineState.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        public void Handle() 
+        {
+            
+        }
+
+        public void SetContext(IStateMachineContext context)
+        {
+            _context = (StateMachineContext)context;
```
Comment (elkavitaly) [inline OneSchedule.Domain/StateMachine/StateMachineState.cs]: Why do you use casting here?

File: OneSchedule/Controllers/TestController.cs [hunk 8]
```diff
@@ -1,70 +1,47 @@
...
-            var existUser = _service.FindFirstAsync(u => u.FirstName == name).Result;
-            if (existUser != null)
-            {
-                existUser.FirstName = user.FirstName;
-                existUser.LastName = user.LastName;
-                existUser.UserName = user.UserName;
-                _service.UpdateAsync(existUser);
-            }
+            return null;
+            throw new BotAppInternalException("This is bot internal app exception");
```
Comment (elkavitaly) [inline OneSchedule/Controllers/TestController.cs]: What is this?

File: OneSchedule.Domain/StateMachine/StateMachineState.cs [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+{
+    public class StateMachineState : IStateMachineState
+    {
+        private StateMachineContext _context;
+
+        public StateMachineState()
+        {
+        }
+
+        public void Handle()
```
Comment (elkavitaly) [inline OneSchedule.Domain/StateMachine/StateMachineState.cs]: You didn't provide an example of how one state can be moved to another one. Please do that and you'll see that some additional changes are required here

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: OneSchedule/Controllers/TestController.cs (hunk 8 above)
Comment (gitrepository100) [reply OneSchedule/Controllers/TestController.cs]: was a test of middleware

**Pull request closed without merging**